pkg/translator/prometheusremotewrite/helper.go

```diff
@@ -96,46 +97,40 @@ var seps = []byte{'\xff'}
 // createAttributes creates a slice of Prometheus Labels with OTLP attributes and pairs of string values.
 // Unpaired string values are ignored. String pairs overwrite OTLP labels if collisions happen and
 // if logOnOverwrite is true, the overwrite is logged. Resulting label names are sanitized.
-func createAttributes(resource pcommon.Resource, attributes pcommon.Map, externalLabels map[string]string,
+func (c *PrometheusConverter) createAttributes(resource pcommon.Resource, attributes pcommon.Map, externalLabels map[string]string,
 	ignoreAttrs []string, logOnOverwrite bool, extras ...string) []prompb.Label {
 	resourceAttrs := resource.Attributes()
 	serviceName, haveServiceName := resourceAttrs.Get(conventions.AttributeServiceName)
 	instance, haveInstanceID := resourceAttrs.Get(conventions.AttributeServiceInstanceID)
 
-	// Calculate the maximum possible number of labels we could return so we can preallocate l
-	maxLabelCount := attributes.Len() + len(externalLabels) + len(extras)/2
-
-	if haveServiceName {
-		maxLabelCount++
-	}
-
-	if haveInstanceID {
-		maxLabelCount++
-	}
-
 	// map ensures no duplicate label name
-	l := make(map[string]string, maxLabelCount)
+	l := c.labelsMap
+	clear(l)
```

Comment:
Could we also clear `labelMap` when we call `reset()`? I think it would be cleaner if we reset the state in one single place instead, or is there any particular reason to do it here?

Reply:
That makes sense, thank you! Could we add a comment explaining it?